### DISCUSSION TITLE
Life after Nextflow+Apache Ignite. Alternatives?

### DISCUSSION THREAD

**Discussion by dgarrimar** [nextflow-io/nextflow#5385] [Q&A]
Hi!

A very simple, widely used Nextflow pipeline is the following:

1) Split a large input file into chunks
2) Run a specific tool for each chunk
3) Collect the results again in the output file

The speed-up achieved increases -obviously- with the number of chunks that can run in parallel (i.e. in an HPC environment, the number of cores/cpus available).

However, in many HPC clusters (e.g. MareNostrum), the maximum number of jobs that can be running/queuing simultaneously is limited (which is reasonable, they aim to schedule large, highly parallelized jobs).
 
Until recently, I used Nextflow + Ignite to "convert" a large set of jobs (e.g. 10000 jobs using 1 CPU each) to a single job (e.g. 1 job using 10000 CPUs). Ignite was "virtualizing" a cluster using the resources and internally sending the jobs one by one (1 job/cpu). 

As Ignite is not supported anymore in NF, and working with very old NF versions that include it starts to give problems too, **I wonder whether there is an alternative in current NF versions to do this specific task.** 

Thanks a lot!

Diego

**Comment by bentsherman** [comment]
The [nf-ignite](https://github.com/nextflow-io/nf-ignite) still exists and could be maintained if people are willing to maintain it.

Aside from that, there is the [array](https://nextflow.io/docs/latest/reference/process.html#array) directive for submitting job arrays, and #3909 will add a similar `batch` directive for task batching, which I think is closer to what you're asking

**Reply by pditommaso** [reply]
Array does not help with HCP like Marenostrum which requires re-allocating the nodes (MPI style)

**Reply by dgarrimar** [reply]
After some suggestions in Slack (thanks to @ewels) and some testing (with *a lot* of help from @emi80), it seems that [HyperQueue ](https://www.nextflow.io/docs/latest/executor.html#hyperqueue) does a really good job to address this type of scenario!. It also has the advantage of fewer dependencies/easier installation and usage with respect to [Flux](https://www.nextflow.io/docs/latest/executor.html#flux-executor), particularly in MN5 nodes without internet access.

**Reply by pditommaso** [reply]
Interesting! I read nice thing about HyperQueue, great to hear it's working in the BSC scenario
